pycbc/workflow/core.py

```diff
@@ -562,18 +567,24 @@ def update_current_tags(self, tags):
 
         # collect the options and profile information
         # from the ini file section(s)
+        self.all_added_options = []
```

Comment:
Make this a set not a list. All you're doing is `is XX in this_list` and that is *waaaaayyyyy* faster with sets (a dictionary's keys works the same as a set, but we don't have values here, so don't want a dict).